### PR TITLE
feat(build):  bundling with rollup, use headless chrome for ci

### PR DESCRIPTION
**Pull request by zxc0328**
+ Update gulpfile and travisci config file to use headless chrome for e2e test.
+ Use rollup for bundling, which reduced minified bundle size to 66kb(5kb smaller).
+ Inject` process.env.NODE_ENV ` when bundling, which allows writing code dedicated for development(concise warnning for developer,  internal performance test , etc). Those code will be removed in the production build.
+ Remove redundant code in gulpfile.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,15 +19,17 @@
   },
   "homepage": "http://regularjs.github.io",
   "scripts": {
-    "test": "gulp travis --phantomjs",
-    "prepublish": "gulp build"
+    "watch": "gulp watch",
+    "test": "gulp travis --chromeHeadless",
+    "prepublish": "gulp build --production"
```
Comment (fengzilong) [inline package.json]: prepublish这个要不要换成prepublishOnly
https://github.com/npm/npm/issues/10074

**Comment by leeluolee** [comment]
赞赞，  算与时俱进了一把， 看了ci都通过了.   不过发现我可能在某个版本把coverage 搞坏了， 后面再修了。

**Comment by leeluolee** [comment]
@zxc0328  发现在IE9- testcase 报错了  .   看了下  testbundle 还是走的webpack。 我相应的做下修改

**Comment by leeluolee** [comment]
先回退了一把，后面再改吧，看了下影响还比较大

**Pull request merged**